explorer/e2e/testFunctions.ts

```diff
@@ -1052,6 +1063,93 @@ export async function testBackpageDetails(
+  tab: TabDescription
+): Promise<boolean> {
+  if (tab?.indexExportPage === undefined) {
+    console.log(
+      "testIndexExportWorkflow Error: indexExportPage not specified for given tab, so test cannot run"
+    );
+    return false;
+  }
+  await page.goto(tab.url);
+  const exportButtonLocator = page.getByRole("link", {
+    name: tab.indexExportPage.indexExportButtonText,
+  });
+  await expect(exportButtonLocator).toBeVisible();
+  await exportButtonLocator.click();
+  await expect(
+    page.getByText(tab.indexExportPage.firstLandingMessage ?? "")
+  ).toBeVisible();
+  await expect(
+    page.getByRole("link", {
+      name: tab.indexExportPage.exportOptionButtonText,
+    })
+  ).toBeEnabled();
+  await page
+    .getByRole("link", { name: tab.indexExportPage.exportOptionButtonText })
+    .click();
+  const exportRequestButtonLocator = page.getByRole("button", {
+    name: tab.indexExportPage.exportRequestButtonText,
+  });
+  // Complete the export request form
+  await makeMinimalExportRequest(page, exportRequestButtonLocator);
+  // Click the Export Action button and check that a download occurs
+  const exportActionButtonLocator = page.getByRole("link", {
+    name: tab.indexExportPage?.exportActionButtonText,
+  });
+  await expect(exportActionButtonLocator).toBeEnabled();
+  const downloadPromise = page.waitForEvent("download", { timeout: 10000 }); // This timeout is necessary, as otherwise the test will wait for the global test timeout
+  await exportActionButtonLocator.click();
+  const download = await downloadPromise;
+  // Cancel the download when it occurs, since there's no need to let it fully download
+  await download.cancel();
+  return true;
+}
+
+export async function testIndexExportSummary(
+  page: Page,
+  tab: TabDescription
+): Promise<boolean> {
+  if (tab?.indexExportPage === undefined) {
+    console.log(
+      "testIndexExportDetails Error: indexExportPage not specified for given tab, so test cannot run"
+    );
+    return false;
+  }
+  await page.goto(tab.url);
+  //await expect(getFirstRowNthColumnCellLocator(page, 0)).toBeVisible();
+  const headers: { header: string; value: string }[] = [];
+  const indexExportButtonLocator = page.getByRole("link", {
+    name: tab.indexExportPage.indexExportButtonText,
+  });
+  await expect(indexExportButtonLocator).toBeVisible();
+  for (const detail of tab.indexExportPage.detailsToCheck) {
+    // This Regexp gets a decimal number, some whitespace, then the name of the detail, matching how the detail box appears to Playwright.
+    const detailBoxRegexp = RegExp(`^([0-9]+\\.[0-9]+k)\\s*${detail}$`);
+    console.log(await page.getByText(detailBoxRegexp).innerText());
+    // This gets the detail's value. The .trim() is necessary since innertext adds extraneous whitespace on Webkit
+    headers.push({
+      header: detail,
+      value: ((await page.getByText(detailBoxRegexp).innerText())
+        .trim()
+        .match(detailBoxRegexp) ?? ["", "ERROR"])[1],
```

Comment:
This is an interesting one! I feel like it'd be slightly more readable if you handled the error case separately/explicitly rather than creating a dummy match array. I'd probably write it more like:

- Get the inner text, trim, match
- If the match is truthy, push the `[1]` value to the headers array
- Otherwise, push "`ERROR`" to the headers array

Thoughts?

Reply:
I refactored it so that it just fails the test if the regex doesn't match fully, let me know if that works! I think it's a little easier to read this way.

Reply:
Got it. I think the regex complexities should be resolved when we started adding test IDs.